src/detect-engine-threshold.c

```diff
@@ -300,7 +300,7 @@ static inline void RateFilterSetAction(Packet *p, PacketAlert *pa, uint8_t new_a
             pa->flags |= PACKET_ALERT_RATE_FILTER_MODIFIED;
             break;
         case TH_ACTION_DROP:
-            PacketDrop(p, PKT_DROP_REASON_RULES_THRESHOLD);
+            PacketDrop(p, new_action, PKT_DROP_REASON_RULES_THRESHOLD);
```

Comment:
new_action seems to have a different set of flags. It has `TH_ACTION_DROP`, but is treated here as if it has `ACTION_DROP`. Even if these would be the same a safer approach would be to just pass `ACTION_DROP` here explicitly.

Reply:
Btw the reject case below should also act like reject+drop, can you review that it does?